### PR TITLE
Zi/server tweaks

### PR DESCRIPTION
**Pull request by lziest**
Enable gokeyless server to listen on Unix socket, which so is client talking to

### CODE REVIEW

File: client/remote_test.go [hunk 1]
```diff
@@ -121,6 +128,21 @@ func TestRemoteGroup(t *testing.T) {
 	}
 }
 
+func TestUnixRemote(t *testing.T) {
+	r, _ := UnixRemote(socketAddr, "localhost")
```
Comment (kisom) [inline client/remote_test.go]: The error should be checked here.

File: server/server.go [hunk 2]
```diff
@@ -416,7 +416,20 @@ func (s *Server) ListenAndServe() error {
...
+	if s.UnixAddr != "" {
+		l, err := net.Listen("unix", s.UnixAddr)
+		if err != nil {
+			return err
+		}
+
+		log.Infof("Listening at unix://%s\n", l.Addr())
+		return s.Serve(l)
+	}
+	return errors.New("can't listen on emtpy path")
```
Comment (kisom) [inline server/server.go]: Typo: "emtpy" → "empty".

Comment (lziest) [inline server/server.go]: thanks, all fixed.

File: client/remote_test.go [hunk 3]
```diff
@@ -121,6 +128,24 @@ func TestRemoteGroup(t *testing.T) {
...
+		t.Fatal(err)
+	}
+	// clear cached remotes and set a unix remote for the client
+	c.DefaultRemote = r
+	c.servers = map[string]Remote{}
+	c.remotes = map[gokeyless.SKI]Remote{}
+
+	// register the ECDDSA certificate again with the default remote
+	registerCertFile(ecdsaPubKey, t)
+	_, err := c.Dial(ecdsaSKI)
```
Comment (kisom) [inline client/remote_test.go]: This is causing the build to fail; should be

```
_, err = c.Dial(ecdsaSKI)
```

Comment (lziest) [inline client/remote_test.go]: my bad

**Comment by kisom** [comment]
LGTM.

**Pull request merged**